src/IdToken.ts

```diff
@@ -89,4 +90,29 @@ export class IdToken {
     }
   }
 
+  static getClaim(claimName: string, rawIdToken: string, logger: Logger): string {
+    if (Utils.isEmpty(rawIdToken)) {
+      throw new Error("null or empty raw idtoken");
+    }
+    
+    const decodedIdToken = Utils.extractIdToken(rawIdToken);
+
+    if (!decodedIdToken)
+      return null;
+
+    return IdToken.getClaimInternal(claimName, decodedIdToken, logger);
+  }
+
+  private static getClaimInternal(claimName: string, decodedIdToken: any, logger: Logger) {
+    let res: string = null;
+
+    if (decodedIdToken.hasOwnProperty(claimName)) {
+      res = decodedIdToken[claimName];
+    }
+    else{
+      logger.warning('unable to find claim: ' + claimName);
```

Comment:
You might want to handle the case when logger is null.

Reply:
absolutely, thanks @saraford !